### PR TITLE
Update Assume Role docs and tests

### PR DESCRIPTION
**Pull request by JordonPhillips**
This updates the assume role documentation and adds tests in response
to the assume role changes in boto/botocore#1313

It additionally adds changelog entries that will result in a minor
version bump.

### CODE REVIEW

File: awscli/topics/config-vars.rst [hunk 1]
```diff
@@ -200,8 +200,18 @@ You can specify the following configuration values for configuring an IAM role
 in the AWS CLI config file:
 
 * ``role_arn`` - The ARN of the role you want to assume.
-* ``source_profile`` - The AWS CLI profile that contains credentials we should
-  use for the initial ``assume-role`` call.
+* ``source_profile`` - The AWS CLI profile that contains credentials /
+  configuration we should use for the initial ``assume-role`` call. This
```
Comment (kyleknap) [inline awscli/topics/config-vars.rst]: Probably want to remove the use of "we" to make it more third person or instructive.

File: awscli/topics/config-vars.rst [hunk 2]
```diff
@@ -200,8 +200,18 @@ You can specify the following configuration values for configuring an IAM role
...
+  profile may be another profile configured to use ``assume-role``, though
+  if static credentials are present in the profile they will be preferred.
```
Comment (kyleknap) [inline awscli/topics/config-vars.rst]: Maybe rephrase this from "preferred" to something a long the lines of taking precedence over `role_arn`/`credential_source`

File: awscli/topics/config-vars.rst [hunk 3]
```diff
@@ -200,8 +200,18 @@ You can specify the following configuration values for configuring an IAM role
...
+  This parameter cannot be provided alongside ``credential_source``.
+* ``credential_source`` - The credential provider to use to get credentials for
+  the initial ``assume-role`` call. This parameter cannot be provided
+  alongside ``source_profile``. Valid values are:
+  * ``Environment`` to pull source credentials from environment variables.
+  * ``Ec2InstanceMetadta`` to use the EC2 instance role as source credentials.
```
Comment (kyleknap) [inline awscli/topics/config-vars.rst]: Ec2InstanceMetadta -> Ec2InstanceMetadata

File: awscli/topics/config-vars.rst [hunk 4]
```diff
@@ -245,6 +255,13 @@ Example configuration::
   role_arn=arn:aws:iam:...
   source_profile=development
 
+Example configuration using ``credential_source``::
+
+  # In ~/.aws/config
+  [profile crossaccount]
+  role_arn=arn:aws:iam:...
+  credential_source=Ec2InstanceMetadata
+
```
Comment (kyleknap) [inline awscli/topics/config-vars.rst]: I wonder if it makes sense to explain what this example does exactly in obtaining credentials?

**Comment by JordonPhillips** [comment]
Cool. I squashed that extra doc commit. This will be merged alongside the relevant botocore pr.
